common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -840,6 +848,61 @@ def _verify_branch_setting(self, expected_branch_setting):
                 actual_setting=actual_branch_setting
             )
 
+    def _compute_course_structure(self, course_id):
```

Comment:
@ormsbee This method should address your concerns regarding inheritance. It is similar to `_compute_metadata_inheritance_tree()` in `base.py`. The differences are as follows:

* Retrieves all nodes instead of non-leaves,
* Retrieves far fewer fields
* Ignores inheritance
* Includes a hack for video and HTML modules (which is really only present to make the unit tests pass).